src/aosm/azext_aosm/common/registry.py

```diff
@@ -0,0 +1,497 @@
+                " push it to the Artifact Store using manifest credentials scoped"
+                " only to the store. This requires Docker to be installed"
+                " locally."
+            )
+
+        try:
+            acr_import_image_cmd = [
+                str(shutil.which("az")),
+                "acr",
+                "import",
+                "--name",
+                target_acr,
+                "--source",
+                source_image,
+                "--image",
+                f"{image_name}:{image_version}",
+                "--password",
+                access_token,
+            ]
+            call_subprocess_raise_output(acr_import_image_cmd)
+        except CLIError as error:
+            logger.debug(error, exc_info=True)
+            if (" 401" in str(error)) or ("Unauthorized" in str(error)):
+                # As we shell out the the subprocess, I think checking for these strings
+                # is the best check we can do for permission failures.
+                raise BadRequestError(
+                    "Failed to import image.\nThe problem may be one or more of:\n"
+                    " - the image_source in your config file does not exist;\n"
+                    " - the image doesn't exist;\n"
+                    " - you do not have permissions to import images.\n"
+                    f"You need to have Reader/AcrPull from {self.registry_name}, "
+                    "and Contributor role + AcrPush role, or a custom "
+                    "role that allows the importImage action and AcrPush over the "
+                    "whole subscription in order to be able to import to the new "
+                    "Artifact store. More information is available at "
+                    "https://aka.ms/acr/authorization\n\n"
+                    "If you do not have the latter then you can re-run the command using "
+                    "the --no-subscription-permissions flag to pull the image to your "
+                    "local machine and then push it to the Artifact Store using manifest "
+                    "credentials scoped only to the store. This requires Docker to be "
+                    "installed locally."
+                ) from error
+
+            # Otherwise, the most likely failure is that the image already exists in the artifact
+            # store, so don't fail at this stage, log the error.
+            logger.warning(
+                (
+                    "Failed to import %s to %s. If this failure is because it already exists in "
+                    "the target registry we can continue. If the failure was for another reason, "
+                    " for example it does not exist in the source registry, this is likely "
+                    "fatal. Attempting to continue.\n"
+                    "%s"
+                ),
+                source_image,
+                target_acr,
+                error,
+            )
```

Comment:
```suggestion
                (
                    "Failed to copy %s from %s to %s. If this failure is because it already exists in "
                    "the target registry we can continue. If the failure was for another reason, "
                    " for example it does not exist in the source registry, this is likely "
                    "fatal. Attempting to continue.\n"
                    "%s"
                ),
                source_image,
                self.registry_name,
                target_acr,
                error,
            )
```